### PR TITLE
chore(developer): fixup references in source files to /developer/src 🎢

### PR DESCRIPTION
**Pull request by mcdurdin**
Moves source for Developer from windows/src/developer to developer/src.

This may be best reviewed commit-by-commit. The changes are mostly tweaks to paths, with the following additions:

* developer/src/Makefile has split the Developer release build from Windows release build
* developer/src/test/auto/Makefile cuts the Developer tests out of the Windows tests
* developer/src/kmndebug is old and irrelevant and therefore deleted
* resources/build/ contains updates to build triggers

Note that the many references to windows/src/global will be changing in the next PR chain!

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/inst/kmdev.wxs [hunk 1]
```diff
@@ -257,35 +252,29 @@
       </Component>
 
       <Component Directory="FontsFolder">
-        <File Name='keymanweb-osk.ttf' TrueType='yes' Source='..\..\developer\tike\xml\layoutbuilder\keymanweb-osk.ttf' KeyPath="yes" />
-      </Component>
-
-      <!-- Redistributable files -->
-
-      <Component Directory="dirRedistSetup">
-        <File Name="setup.exe" Source="..\..\..\bin\desktop\setup-redist.exe" KeyPath="yes" />
```
Comment (mcdurdin) [inline developer/src/inst/kmdev.wxs:266]: I have removed the unsigned Keyman for Windows setup.exe that used to be distributed as part of Keyman Developer from the Developer build and it is now in the Windows release build as setup-redist.exe, for use in bundling (see the Keyman for Windows TC build configuration)

File: windows/src/test/unit-tests/Makefile [hunk 2]
```diff
@@ -18,6 +13,16 @@ test:
 
 # ----------------------------------------------------------------------
 
+# TODO: move shared-tests to /common
```
Comment (jahorton) [inline windows/src/test/unit-tests/Makefile:16]: Related note:  I noticed that a lot of keyboard testing resources are getting shuffled around... but to subfolders of `developer`.  I suppose this note is also with respect to those file movements?

**Review by jahorton — COMMENTED** [review]
I'm going to stop my review here for now - there's a _lot_ going on with Makefiles that's hard to fully trace out.

That _may_ be because I'm looking at all commits at once, but either way... there are a fair few things I wanted to at least make note of already.

File: developer/src/Makefile [hunk 3]
```diff
@@ -11,66 +11,199 @@ TARGETS=kmcmpdll kmcomp kmanalyze kmconvert tike samples setup inst kmdecomp
...
-    cd $(ROOT)\src\developer\inst
+    cd $(DEVELOPER_ROOT)\src\inst
     $(MAKE) $(TARGET)
 
+# ----------------------------------------------------------------------
+# ----------------------------------------------------------------------
+# release
+# ----------------------------------------------------------------------
+
+release: clean-pathdefines test-release-cef-path test-releasebuildcheck test-releaseexists test-cert test-7zip test-wix clean clean-output
```
[the remark this answers is not in the document]
Comment (jahorton) [reply developer/src/Makefile:82]: On my second pass, I noticed that the bulk of these changes came in the commit https://github.com/keymanapp/keyman/pull/6641/commits/433c655c476f2e78a2fa2633a0e02a7dc9b393da.  There wasn't much corresponding removed code elsewhere in the commit, though - at least so far as I can tell, anyway.

File: developer/src/samples/Makefile [hunk 4]
```diff
@@ -3,13 +3,13 @@
 #
 
 TARGETS=imsample
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/samples/Makefile:5]: Except that this is a sample project *for* Keyman Developer that is deployed as part of the install package -- unlike samples for Keyman Engine which are only in-repo. So in this case it does belong in the `src` folder. 😄

File: developer/src/Defines.mak [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+DEVELOPER_ROOT=$(KEYMAN_ROOT)\developer
+DEVELOPER_PROGRAM=$(DEVELOPER_ROOT)\bin
+DEVELOPER_OUTLIB=$(DEVELOPER_ROOT)\lib
+DEVELOPER_DEBUGPATH=$(DEVELOPER_ROOT)\debug
+
+KEYMAN_MODELCOMPILER_ROOT=$(DEVELOPER_ROOT)\src\kmlmc
+
+# Temporary import of windows/src/Defines.mak
+!include $(WINDOWS_ROOT)\src\Defines.mak
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/Defines.mak:16]: Added as a TODO to #5816.

File: developer/src/Header.mak [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+WINDOWS_ROOT=$(KEYMAN_ROOT)\windows
+WINDOWS_PROGRAM=$(WINDOWS_ROOT)\bin
+
+DEVELOPER_ROOT=$(KEYMAN_ROOT)\developer
+DEVELOPER_PROGRAM=$(DEVELOPER_ROOT)\bin
+DEVELOPER_OUTLIB=$(DEVELOPER_ROOT)\lib
+DEVELOPER_DEBUGPATH=$(DEVELOPER_ROOT)\debug
+
+# Temporary import of windows/src/Header.mak
+!include $(WINDOWS_ROOT)\src\Header.mak
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/Header.mak:14]: Added as a TODO to #5816.

File: developer/src/Makefile (hunk 3 above)
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/Makefile:82]: Part of the trickiness here is that splitting the build for Developer and Windows means that the Developer release build now has to do some of the release build steps that were previously only done in the Windows build -- but not all of them!

File: developer/src/Makefile [hunk 7]
```diff
@@ -11,66 +11,199 @@ TARGETS=kmcmpdll kmcomp kmanalyze kmconvert tike samples setup inst kmdecomp
 MANIFESTS=kmcomp tike setup
 CLEANS=clean-developer
 
-!include ..\Header.mak
+EXCLUDEPATHDEFINES=1
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/Makefile:14]: 👀 

That work for you?

File: developer/.gitignore [hunk 8]
```diff
@@ -0,0 +1,5 @@
+bin/
+build/
+debug/
+lib/
+release/
```
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/.gitignore:5]: Not sure I agree with your reorg thoughts here.

Keyman Developer will still be a single product, even if it has cross-platform deployment targets. Note that developer/kmlmc and developer/server moved to src/ in #6630 and #6643, so all Developer modules are under developer/src. This gives us a final structure of:

```
developer\src
  common
  ext
  images
  inst
  kmanalyze
  kmcmpdll
  kmcomp
  kmconvert
  kmdecomp
  kmlmc
  samples
  server
  setup
  test
  tike
  tools
```

Most of those folders are modules. The ones that aren't, **common**, **ext**, **images**, **test**, and **tools** are fairly obvious.

We could go and refactor further to move **inst**, **kmanalyze**, **kmcmpdll**, **kmcomp**, **kmconvert**, **kmdecomp**, **kmlmc**, **server**, **setup** and **tike** into a different parent folder, but not sure where that really gets us.

In the end, the important thing I am trying to accomplish here is that Developer does not have dependencies on Windows, and can be built independently. Like each of the other products, it is a top-level folder.

As we start to make Developer more cross-platform, we can always revisit the internal structure of the developer tree, but restructuring developer/ really wasn't a goal of this refactor anyway.

File: windows/src/test/unit-tests/Makefile (hunk 2 above)
Comment (mcdurdin) [reply windows/src/test/unit-tests/Makefile:16]: The keyboard testing resources here are **compiler** tests, not **keyboard** tests. Therefore, those resources belong in the developer unit tests folder (where they will be moved in a later PR). There are other unit tests that are testing shared code, such as those moved in #6758.

File: developer/.gitignore (hunk 8 above)
[the remark this answers is not in the document]
Comment (jahorton) [reply developer/.gitignore:5]: Re the #6630 part:  I had a comment there (https://github.com/keymanapp/keyman/pull/6630#discussion_r894133545) written up at this point already; I'd simply forgotten to click the "submit" button.  (I was wondering why it wasn't mentioned, then realized.)

File: developer/src/Makefile (hunk 7 above)
[the remark this answers is not in the document]
Comment (mcdurdin) [reply developer/src/Makefile:14]: @ermshiperete do you want to take a look at this one as well?

**Pull request merged**